syntax = "proto2";

package lego.nat.protobuf;

message DetectionRequest {
	optional bytes public_ip = 1;
    optional int32 public_port = 2;
    optional bytes local_ip = 3;
    optional int32 local_port = 4;
    optional bytes id = 5;
    optional int32 nat_type = 6;
    optional bytes dht_key = 7;
	optional bool client = 8;
    optional uint32 min_svr_port = 9;
    optional uint32 max_svr_port = 10;
    optional uint32 min_route_port = 11;
    optional uint32 max_route_port = 12;
    optional uint32 node_weight = 13;
    optional uint32 min_udp_port = 14;
    optional uint32 max_udp_port = 15;
    optional bytes node_tag = 16;
}

message NatMessage {
	optional DetectionRequest detection_req = 1;
}